mkosi/initrd.py

```diff
@@ -88,89 +89,99 @@ def main() -> None:
-                shutil.copy2(Path("/etc") / p, Path(d) / "etc" / p)
+            if Path(d).exists():
+                cmdline += ["--include", d]
+
+        with tempfile.TemporaryDirectory() as d:
+            # Make sure we don't use any of mkosi's default repositories.
+            for p in (
+                "yum.repos.d/mkosi.repo",
+                "apt/sources.list.d/mkosi.sources",
+                "zypp/repos.d/mkosi.repo",
+                "pacman.conf",
+            ):
+                (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+                (Path(d) / "etc" / p).touch()
+
+            # Copy in the host's package manager configuration.
+            for p in (
+                "dnf",
+                "yum.repos.d/",
+                "apt",
+                "zypp",
+                "pacman.conf",
+                "pacman.d/",
+            ):
+                if not (Path("/etc") / p).exists():
+                    continue
+
+                (Path(d) / "etc" / p).parent.mkdir(parents=True, exist_ok=True)
+                if (Path("/etc") / p).resolve().is_file():
+                    shutil.copy2(Path("/etc") / p, Path(d) / "etc" / p)
+                else:
+                    shutil.copytree(
+                        Path("/etc") / p,
+                        Path(d) / "etc" / p,
+                        ignore=shutil.ignore_patterns("gnupg"),
+                        dirs_exist_ok=True,
+                    )
+
+            cmdline += ["--sandbox-tree", d]
+
+            # Prefer dnf as dnf5 has not yet officially replaced it and there's a much bigger chance that
+            # there will be a populated dnf cache directory.
+            run(
+                cmdline,
+                stdin=sys.stdin,
+                stdout=sys.stdout,
+                env={"MKOSI_DNF": dnf.name} if (dnf := find_binary("dnf", "dnf5")) else {},
+            )
+
+            if not args.output_dir:
+                args.output_dir = Path.cwd()
+
+            log_notice(f"Copying {staging_dir}/{args.output} to {args.output_dir}/{args.output}")
+            if args.format == OutputFormat.directory.value:
+                copy_tree(Path(staging_dir), Path(args.output_dir), preserve=False)
             else:
-                shutil.copytree(
-                    Path("/etc") / p,
-                    Path(d) / "etc" / p,
-                    ignore=shutil.ignore_patterns("gnupg"),
-                    dirs_exist_ok=True,
-                )
-
-        cmdline += ["--sandbox-tree", d]
-
-        # Prefer dnf as dnf5 has not yet officially replaced it and there's a much bigger chance that there
-        # will be a populated dnf cache directory.
-        run(
-            cmdline,
-            stdin=sys.stdin,
-            stdout=sys.stdout,
-            env={"MKOSI_DNF": dnf.name} if (dnf := find_binary("dnf", "dnf5")) else {},
-        )
+                shutil.copy2(Path(staging_dir) / args.output, Path(args.output_dir))
```

Comment:
Doesn't this copy only the symlink instead?

Reply:
Ah I guess it dereferences it

Reply:
Can we make that explicit by adding a `resolve()` here and a comment?

Reply:
And in that case, we can use `copy_tree()` in both cases as well

Reply:
hmm, not on my tests... but I actually forgot to create the output directory if it does not exist

Reply:
Done and tested ok.